packages/yourloops/lib/auth/user-api.ts

```diff
@@ -0,0 +1,70 @@
+import { Preferences, Profile, Settings } from "../../models/shoreline";
+import HttpService from "../../services/http";
+import { HttpHeaderKeys } from "../../models/api";
+
+export default class UserApi {
+
+  static async getShorelineAccessToken(email: string): Promise<string> {
+    try {
+      const { headers } = await HttpService.post({ url: `auth/hack/user/${email}` });
+      return headers[HttpHeaderKeys.sessionToken];
+    } catch (err) {
+      console.log("This profile doesn't exists");
+      return "no-token";
```

Comment:
Shouldn't we throw an error instead?

Reply:
This function is here temporarily just to fetch old token for backend services that are not yet auth0 compatible. If throwing an error we have to handle this in the auth hook instead, but the result will be the same.

Reply:
I think that throwing an error is always cleaner to handle incorrect behaviour. So it mainly depends on whereas the code will run without further issues if the session token value is "no-token"? Because currently, this code suggest that having session token set to "no-token" is fine. I doubt that it is? If it is fine, then it is not easy to understand the rest of the code (mainly the hook). Otherwise, it should be handled more properly by throwing an error.

Reply:
And I get that this code is temporary, if you think that handling this case by throwing an error is too expensive, then don't do it. But then it must be fixed in a near future

Reply:
Yes this code will be removed as soos as all backend services will be compatible with Auth0. There is a pending ticket [YLP-1553](https://diabeloop.atlassian.net/browse/YLP-1553)